docs/generator_template.md

```diff
@@ -0,0 +1,569 @@
+
+```cmd
+Generator Error: This template requires the following missing params: server.
+```
+
+Update your `test:generate` script in `package.json` to include the server param `test:generate": "asyncapi generate fromTemplate test/fixtures/asyncapi.yml ./ --output test/project --force-write --param server=dev "`
+You can now replace the static broker from `mqttBroker = 'test.mosquitto.org'` to `mqttBroker = "${asyncapi.servers().get(params.server).url()}"` in **index.js**.
+
+Now the template code looks like this:
+
+``` js
+import { File } from '@asyncapi/generator-react-sdk';
+
+// notice that now the template not only gets the instance of parsed AsyncAPI document but also the parameters
+export default function ({ asyncapi, params }) {
+ 
+  return (
+    <File name="client.py">
+      {`import paho.mqtt.client as mqtt
+
+mqttBroker = "${asyncapi.servers().get(params.server).url()}"
+
+class TemperatureServiceClient:
+    def __init__(self):
+        self.client = mqtt.Client()
+        self.client.connect(mqttBroker)
+
+
+    def sendTemperatureChange(self, id):
+        topic = "temperature/changed"
+        self.client.publish(topic, id)`}
+    </File>
+  )
+}
+```
+
+Run `npm test` to validate that your code still works as expected.
+
+### Templating index.js with React
+
+Python takes indentation very seriously, and our generated output will be Python code. We, therefore, need to make sure the indentation in index.js looks right so the generated code is indented correctly. After templating the code in index.js, it will look like the following code snippet:
+
+```js
+// 1
+import { File, Text } from '@asyncapi/generator-react-sdk'
+export default function ({ asyncapi, params }) {
+  return (
+    <File name="client.py">
+    // 2
+      <Text newLines={2}>import paho.mqtt.client as mqtt</Text>
+    // 3
+      <Text newLines={2}>mqttBroker = "{asyncapi.servers().get(params.server).url()}"</Text>
+    // 4
+      <Text newLines={2}>class {asyncapi.info().title().replaceAll(' ', '')}Client:</Text>
+    // 5
+      <Text indent={2} newLines={2}>
+        {`def __init__(self):
+            self.client = mqtt.Client()
+            self.client.connect(mqttBroker)`}
+      </Text>
+      </File>
+  )
+}
+```
+
+1. Import the **Text** component that will wrap strings so they are indented properly in the output. Your import statement should now look like this: `import { File, Text } from '@asyncapi/generator-react-sdk'`.
+2. When the paho module import is rendered in **client.py** file, it will add two extra new lines.
+3. The broker url is templated in a Text component removing the `$` from the string template.
+4. Dynamically get the class name **TemperatureServiceClient** from the AsyncAPI document from the **info** object using the Parser API using the code: `asyncapi.info().title()` . It will return `Temperature Service`, then remove the spaces and add Client as a suffix.
```

Comment:
```suggestion
4. Dynamically get the class name **TemperatureServiceClient** from the AsyncAPI document from the **info** object using the Parser API using the code: `asyncapi.info().title()` . It will return `Temperature Service`, then remove the spaces and add `Client` as a suffix.
```